src/lib/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -910,6 +911,50 @@ public function loadVersionInfo(int $contentId, ?int $versionNo = null): array
         return $queryBuilder->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    public function loadVersionNoArchivedWithin(int $contentId, int $seconds): array
+    {
+        $now = new \DateTime();
+        $interval = DateInterval::createFromDateString($seconds . ' seconds');
+        if ($interval === false) {
+            return [];
+        }
+        $now->sub($interval);
+        $queryBuilder = $this->queryBuilder->createVersionInfoFindQueryBuilder();
+        $expr = $queryBuilder->expr();
+
+        $queryBuilder
+            ->where(
```

Comment:
I prefer to stay away from `->where()`, because it drops any existing conditions from query without warning.

Consider that this would go unnoticed if `QueryBuilder::createVersionInfoFindQueryBuilder()` added it's own preliminary conditions, which would be overwritten here.

```suggestion
            ->andWhere(
```